Bump ably-go to feature/conn-open-failures-rtn14 (80f6c2a)

lmars opened the pull request:
This updates the ably-go dependency to use the majority of the 1.2 changes plus the changes from the [connection open retries PR](https://github.com/ably/ably-go/pull/172) (which right now is missing some commits from the 1.2 integration branch, but should be good enough for us to test with).

paddybyers reviewed (commented):
Should we then also remove the workarounds that add explicit retries (and instead rely on the library) ?

lmars commented:
> Should we then also remove the workarounds that add explicit retries (and instead rely on the library) ?

Good point, I removed the connection retries in 1bf26d9 (the publish retries remain as they are not yet handled by ably-go, I plan to open a separate issue to deal with that)

I also updated to the 1.2 API in 1d64a82.

lmars commented:
I added 536b974 to workaround [this problem](https://github.com/ably/ably-go/pull/172#issuecomment-718312239).
